tests/common.py

```diff
@@ -0,0 +1,139 @@
+"""
+Contains several commonly used objects (models, dataloaders)
+that are shared across the test suite.
+"""
+from typing import Sequence
+
+import numpy as np
+import torch
+from PIL import Image
+from torch.utils.data import DataLoader, Dataset
+from torchvision.datasets import VisionDataset
+
+from composer.models import ComposerClassifier
+
+
+class SimpleModel(ComposerClassifier):
+    """ Small classification model with 10 input features and 2 classes.
+
+    Args:
+        num_features (int): number of input features (default: 10)
+        num_classes (int): number of classes (default: 2)
+    """
+
+    def __init__(self, num_features: int = 10, num_classes: int = 2) -> None:
+
+        self.num_features = num_features
+        self.num_classes = num_classes
+
+        fc1 = torch.nn.Linear(num_features, 5)
+        fc2 = torch.nn.Linear(5, num_classes)
+
+        net = torch.nn.Sequential(
+            torch.nn.AdaptiveAvgPool2d(1),
+            torch.nn.Flatten(),
+            fc1,
+            torch.nn.ReLU(),
+            fc2,
+            torch.nn.Softmax(dim=-1),
+        )
+        super().__init__(module=net)
+
+        # fc1 and fc2 are bound to the model class
+        # for access during several surgery tests
+        self.fc1 = fc1
+        self.fc2 = fc2
+
+
+class SimpleConvModel(ComposerClassifier):
+    """ Small convolutional classifer
+
+    Args:
+        num_channels (int): number of input channels (default: 32)
+        num_classes (int): number of classes (default: 2)
+    """
+
+    def __init__(self, num_channels: int = 32, num_classes: int = 2) -> None:
+
+        self.num_classes = num_classes
+        self.num_channels = num_channels
+
+        conv_args = dict(kernel_size=(3, 3), padding=1)
+        conv1 = torch.nn.Conv2d(in_channels=num_channels, out_channels=8, **conv_args)
+        conv2 = torch.nn.Conv2d(in_channels=8, out_channels=4, **conv_args)
+        pool = torch.nn.AdaptiveAvgPool2d(1)
+        flatten = torch.nn.Flatten()
+        fc1 = torch.nn.Linear(4, 16)
+        fc2 = torch.nn.Linear(16, num_classes)
+
+        net = torch.nn.Sequential(
+            conv1,
+            conv2,
+            pool,
+            flatten,
+            fc1,
+            fc2,
+        )
+        super().__init__(module=net)
+
+        # bind these to class for access during
+        # surgery tests
+        self.conv1 = conv1
+        self.conv2 = conv2
+
+
+class RandomClassificationDataset(Dataset):
```

Comment:
We opt to use test suite's simpler datasets compared to `SyntheticDatasets` to make the tests simpler to read and debug.

Reply:
Can you add this GH comment into the code?

Reply:
Also, do we need new tests for the synthetic datasets (are any tests still using them?)